### ISSUE TITLE
[FEATURE]

### ISSUE THREAD

**Issue by mmalaiska** [rootiest/printcfg#26]
hi,
I could not get the start gcode to play nice with OrcaSlicer. 

1. Failed to generate G-code for invalid custom G-code.
machine_start_gcode Parsing error at line 2: Not a variable name
SET_GCODE_VARIABLE MACRO=_printcfg VARIABLE=extruder_temp VALUE={first_layer_temperature[initial_extruder] + extruder_temperature_offset[initial_extruder]}
^
It is pointing to extruder_temperature_offset[initial_extruder].
I removed this offset from start gcode. Onto point 2.

2. Failed to generate G-code for invalid custom G-code.
machine_start_gcode Parsing error at line 4: Referencing a vector variable when scalar is expected
SET_GCODE_VARIABLE MACRO=_printcfg VARIABLE=chamber_temp VALUE={chamber_temperature}
                                                                                                                                      ^
It is pointing at {chamber_temperature}. I removed this line. Onto point 3.

3. Failed to generate G-code for invalid custom G-code.
machine_start_gcode Parsing error at line 5: Not a variable name
SET_GCODE_VARIABLE MACRO=_printcfg VARIABLE=material_color VALUE={filament_colour_int}
                                                                                                                                     ^
Now it is pointing at {filament_colour_int}. Also removed the line. Onto point 4.

4. Failed to generate G-code for invalid custom G-code.
machine_start_gcode Parsing error at line 7: Not a variable name
SET_GCODE_VARIABLE MACRO=_printcfg VARIABLE=tool_name VALUE="'{tool_name[initial_extruder]}'"
                                                                                                                                 ^
Now it is pointing at {tool_name[initial_extruder]}.

I have zero understanding in coding macros, or coding of any sort :)
Can you provide start and other gcodes for OrcaSlicer? Can your configuration play nice with it? I have a feeling Orca does not use the above mentioned variables at all... In any case, appreciate your work, thank you

**Comment by rootiest** [comment]
Yeah unfortunately this only works with SuperSlicer (and most should be able to carry over into PrusaSlicer)

PrusaSlicer doesn't support chamber temp, so you will need to manually assign values to that.

As for Orca, the issue is that we are not provided with variables we can use for most of the settings, only the basic temperature variables.

You can see [my zippy-guides Macros guide](https://github.com/rootiest/zippy_guides/blob/main/guides/macros.md) for an overview of the correct variable names in Orca for bed, extruder, and chamber temps, but none of the filament variables are possible there at this time.

**Comment by rootiest** [comment]
So you can use SuperSlicer to get the full functionality, or PrusaSlicer to get everything but chamber temp.

But Orca can only do the temperatures (extruder, bed, chamber) so no filament-related stuff, etc.

**Comment by mmalaiska** [comment]
understood, thank you for a quick answer
